src/commands/mobile_app/upload.rs

```diff
@@ -51,9 +51,44 @@ pub fn make_command(command: Command) -> Command {
                 .required(true),
         )
         .arg(
-            Arg::new("sha")
-                .long("sha")
-                .help("The git commit sha to use for the upload. If not provided, the current commit sha will be used.")
+            Arg::new("head_sha")
+                .long("head-sha")
+                .help("The VCS commit sha to use for the upload. If not provided, the current commit sha will be used.")
+        )
+        .arg(
+            Arg::new("base_sha")
+                .long("base-sha")
+                .help("The VCS commit's base sha to use for the upload. If not provided, the merge-base of the current and remote branch will be used.")
+        )
+        .arg(
+            Arg::new("vcs_provider")
+                .long("vcs-provider")
+                .help("The VCS provider to use for the upload. If not provided, the current provider will be used.")
+        )
+        .arg(
+            Arg::new("head_repo_name")
+                .long("head-repo-name")
+                .help("The name of the git repository to use for the upload (e.g. organization/repository). If not provided, the current repository will be used.")
+        )
+        .arg(
+            Arg::new("base_repo_name")
+                .long("base-repo-name")
+                .help("The name of the git repository to use for the upload (e.g. organization/repository). If not provided, the current repository will be used.")
+        )
+        .arg(
+            Arg::new("head_ref")
+                .long("head-ref")
+                .help("The reference (branch) to use for the upload. If not provided, the current reference will be used.")
+        )
+        .arg(
+            Arg::new("base_ref")
+                .long("base-ref")
+                .help("The reference (branch) to use for the upload. If not provided, the current reference will be used.")
+        )
+        .arg(
+            Arg::new("pr_number")
+                .long("pr-number")
+                .help("The pull request number to use for the upload. If not provided, the current pull request number will be used.")
```

Comment:
This way, Clap can parse straight to a `u32` and perform the validation internally

```suggestion
            Arg::new("pr_number")
                .long("pr-number")
                .value_parser(clap::value_parser!(u32))
                .help("The pull request number to use for the upload. If not provided, the current pull request number will be used.")
```

Reply:
Very nice to know, figured there was a way to do this I was missing